mailbox/caching/src/main/java/org/apache/james/mailbox/caching/CacheLoaderFromUnderlying.java

```diff
@@ -1,5 +1,5 @@
 package org.apache.james.mailbox.caching;
 
-public interface CacheLoaderFromUnderlying<Key, Value, Underlying, Except extends Throwable> {
-    Value load(Key key, Underlying underlying) throws Except;
+public interface CacheLoaderFromUnderlying<KeyT, ValueT, UnderlyingT, ExceptT extends Throwable> {
```

Comment:
Here it is:
```
[ERROR] src/main/java/org/apache/james/mailbox/caching/CacheLoaderFromUnderlying.java:[3,44] (naming) InterfaceTypeParameterName: Interface type name 'Key' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'.
```

Reply:
ok, but is it really a good idea to enforce it? I'm not a big fan of the result.

Reply:
It's not stupid to have a way to know we are talking about types.
We can disable this rule if you don't agree.

Reply:
your IDE shoud be able to show you what is a class, a type parameter, a field, etc. I don't like to need a prefix or a suffix to show that.

Reply:
I would never did that if it were not a widely adopted practice. I really don't care about these kind of convention but if it can ease onboarding, why not ?
We'll discuss that tomorrow to take a decision ?

Reply:
let's keep it